### PR TITLE
Temp8 3298 move rollback

### PR DESCRIPTION
**Pull request by madsrasmussen**
### Prerequisites

- [ ] I have [created an issue](https://github.com/umbraco/Umbraco-CMS/issues) for the proposed changes in this PR, the link is:
- [ ] I have added steps to test this contribution in the description below

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbcontentnodeinfo.directive.js [hunk 1]
```diff
@@ -101,6 +101,20 @@
                 scope.node.template = templateAlias;
             };
 
+            scope.openRollback = function() {
+                console.log("rollback");
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbcontentnodeinfo.directive.js]:
```suggestion
                
```

Remove the console.log

**Comment by warrenbuckley** [comment]
@zpqrtbnk after the various questions today & yesterday can you test this out please

## Test Notes 
- [ ] Code review & sanity check
- [ ] Create variant node with 2 or more languages
- [ ] Publish changes to node. Both variants, one variant, etc...
- [ ] Ensure the rollback dialog on info content app opens & lists the versions of the culture/variant 
- [ ] Pick a rollback & confirm the diff changes
- [ ] Perform the rollback & verify the node is refreshed with the correct changes to the node
- [ ] Check the log messages on disk (txt or json)
- [ ] Check the audit message displays a save & then a rollback message in the history 

##Repeat Tests
- [ ] Try with invariant node
- [ ] Try with complex property - grid
- [ ] Try with complex property - content picker
- [ ] Anything else?

**Comment by zpqrtbnk** [comment]
Have reviewed the server-side part of it, and pushed a commit with some changes, can you have a look @warrenbuckley and make sure it all still works? It's basically getting the versions in a more efficient way. It also will only get the latest 50 versions, otherwise we may be breaking things with tons of versions - and then we may want to paginate somehow to get more versions?

**Comment by zpqrtbnk** [comment]
Have reviewed the server-side part of it, and pushed a commit with some changes, can you have a look @warrenbuckley and make sure it all still works? It's basically getting the versions in a more efficient way. It also will only get the latest 50 versions, otherwise we may be breaking things with tons of versions - and then we may want to paginate somehow to get more versions?

**Comment by warrenbuckley** [comment]
It's all good - YAY merged in 🎉 
YAY

File: src/Umbraco.Web/Editors/ContentController.cs [hunk 2]
```diff
@@ -1702,5 +1702,93 @@ public void PostNotificationOptions(int contentId, [FromUri] string[] notifyOpti
...
+            var version = Services.ContentService.GetVersion(versionId);
+            var content = MapToDisplay(version);
+                       
+			return culture == null
+				? content.Variants.FirstOrDefault()  //No culture set - so this is an invariant node - so just list me the first item in here
+                : content.Variants.FirstOrDefault(x => x.Language.IsoCode == culture);
+        }
+
+        [HttpPost]
+        public HttpResponseMessage PostRollbackContent(int contentId, int versionId, string culture = "*")
```
Comment (Shazwazza) [inline src/Umbraco.Web/Editors/ContentController.cs]: There's a permission check missing here. In order to rollback you need the rollback permission. So should have an attribute: `[EnsureUserPermissionForContent("contentId", 'K')]`. I'll fix up.

**Pull request merged**